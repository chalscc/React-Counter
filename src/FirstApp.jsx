import PropTypes from 'prop-types';


const FirstApp = ( {title, subtitle} ) => {

   return (
    <>
        <h1 data-testid="test-title">{ title }</h1>
        <p>{ subtitle }</p>
    </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

FirstApp.defaultProps = {
    // title: 'There is no title',
    subtitle: 'There is no subtitle'
}

export default FirstApp;