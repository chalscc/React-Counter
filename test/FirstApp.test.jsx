import { render } from "@testing-library/react"
import FirstApp from "../src/FirstApp"

describe('Pruebas en <FirstApp /> component', () => { 
    test('debe de hacer match con el snapshot', () => { 

        const title = 'Hola, soy Carles'

        const { container } = render(<FirstApp title={ title } />)

        // console.log(container)

        expect(container).toMatchSnapshot()
     })

     test('debe de mostrar el titulo en un h1', () => { 

        const title = 'Hola, soy Carles'

        const { container, getByText } = render(<FirstApp title={ title } />)

        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');

        expect(h1.innerHTML).toContain(title)
    })
 })