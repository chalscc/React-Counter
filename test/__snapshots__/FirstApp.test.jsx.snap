// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> component debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Carles
  </h1>
  <p>
    There is no subtitle
  </p>
</div>
`;
